r-package/grf/R/plot.R

```diff
@@ -93,3 +97,76 @@ plot.grf_tree <- function(x, ...) {
   dot_file <- export_graphviz(x)
   DiagrammeR::grViz(dot_file)
 }
+
+leaf_stats <- function(forest, samples) UseMethod("leaf_stats")
+
+
+#' A default leaf_stats for forests classes without a leaf_stats method
+#' that always returns NULL.
+#' @param forest Any forest
+#' @param samples The samples to include in the calculations.
+#' @param ... Additional arguments (currently ignored).
+#'
+#' @return NULL
+#'
+#' @method leaf_stats default
+leaf_stats.default <- function(forest, samples, ...){
+  return(NULL)
+}
+
+#' Calculate summary stats given a set of samples for regression forests.
+#' @param forest The GRF forest
+#' @param samples The samples to include in the calculations.
+#' @param ... Additional arguments (currently ignored).
+#'
+#' @return A named vector containing summary stats
+#'
+#' @method leaf_stats regression_forest
+leaf_stats.regression_forest <- function(forest, samples, ...){
+  leaf_stats <- c()
```

Comment:
I think this could be written as 

```
leaf_stats <- c()
leaf_stats["avg Y"] <- round(mean(forest$Y.orig[samples]), 2)
return(leaf_stats)
```

Reply:
Thanks, I learned a few new things from your comments!